draft-proxied-svcb-headers.md

```diff
@@ -17,132 +17,551 @@ author:
+corresponding request did not include a "Proxy-DNS-Request".
+
+Proxy servers MUST attempt to resolve the SVCB DNS query name to
+obtain the SVCB RRset to return in this header.  If this resolves to a
+SVCB AliasMode record, proxy servers MUST resolve the TargetName of
+that AliasMode record to obtain a ServiceMode record.  Both
+resolutions MUST use the same RR type specified in the "t" parameter
+of Proxy-DNS-Request.
+Which records are returned in this header is determined by:
+
+* If the SVCB DNS query name resolves to a SVCB ServiceMode record,
+  only the ServiceMode records SHALL be included in the Proxy-DNS-SVCB
+  list.
+
+* If the SVCB DNS query name resolves to a SVCB AliasMode record, and
+  if the TargetName of the SVCB AliasMode record resolves to a SVCB
+  ServiceMode RRSet, only the ServiceMode records SHALL be included in
+  the Proxy-DNS-SVCB list.
+  
+* If the SVCB DNS query name resolves to a SVCB AliasMode record, and
+  if the TargetName of a SVCB AliasMode record does not resolve to any
+  ServiceMode records, the Proxy-DNS-SVCB list SHALL include a single
+  item containing the AliasMode TargetName and "priority=0".
+
+* If no SVCB records are found authoritatively (i.e., a "NOERROR" or
+  "NXDOMAIN" response when resolving the SVCB DNS query name specified
+  in the Proxy-DNS-Request header), the list MUST contain a single
+  entry with the item value of ".", indicating that the name has no
+  SVCB records.  This special value of "." MUST NOT be returned in
+  error conditions such as timeouts.  The value of the "ttl" parameter
+  MUST be the TTL of this authoritative DNS response.
+
+* If an error condition is encountered (such as a timeout, loop, or
+  invalid record), the Proxy-SVCB-DNS header MUST contain an empty list
+  (and thus not be returned).
+
+
+### Example Proxy-DNS-SVCB response
+
+As an example, assume that the server resolved the SVCB DNS query name
+"example.com" and RR type 65 (HTTPS) as follows:
 
-Each SvcParam that matches a key requested by the client is a parameters with a key
-that is the character "p" followed by the numeric version of the SvcParamKey. For example,
-the ALPN SvcParamKey, with the numeric value 1, would have a parameter key "p1". The value
-of each parameter MUST be an sf-binary item that contains the bytes of the SvcParamValue.
+~~~ diagram
+   example.com.     7200 IN HTTPS 0 foo.svc.example.net.
+   foo.svc.example.net 1800 IN CNAME svc.example.net.
+   svc.example.net. 3600 IN HTTPS 1 svc2.example.net. alpn=h2,h3 echconfig="123..."
+   svc.example.net. 3600 IN HTTPS 2 . alpn=h2 echconfig="abc..."
+~~~
+
+A successful CONNECT response would include the following headers, if the client requested both
+"alpn" and "echconfig":
+
+~~~ example
+HEADERS
+:method = CONNECT
+:status = 200
+proxy-dns-svcb = "svc2.example.net.";priority=1;ttl=1800;key1=:aDIsaDM=:;key5=:MTIzLi4u:,
+                 "svc.example.net.";priority=2;ttl=1800;key1=:aDI=:;key5=:YWJjLi4u:
```

Comment:
Given that the ellipsis is intended to represent longer content in the field, it might be better not to encode literal `...` in these binary examples.
```suggestion
proxy-dns-svcb = "svc2.example.net.";priority=1;ttl=1800;key1=:aDIsaDM=:;key5=:MTIz...:,
                 "svc.example.net.";priority=2;ttl=1800;key1=:aDI=:;key5=:YWJj...:
```